server/src/tls.rs

```diff
@@ -0,0 +1,360 @@
+use rustls::{ServerConfig,RootCertStore};
+
+use x509_parser::prelude::{X509Certificate, FromDer};
+use x509_parser::oid_registry::OidRegistry;
+
+use std::fs;
+use std::sync::Arc;
+use std::io::BufReader;
+
+use anyhow::{anyhow, bail, Context, Result};
+use common::encoding::{decode_utf16le, encode_utf16le};
+use common::settings::Collector;
+
+use hyper::body::HttpBody;
+use hyper::{Body, Request};
+use crate::sldc;
+use sha1::{Sha1,Digest};
+use hex::ToHex;
+
+use log::{info,debug};
+
+
+/// Load certificates contained inside a PEM file
+fn load_certs(filename: &str) -> Result<Vec<rustls::Certificate>, std::io::Error> {
+    let certfile = fs::File::open(filename)?;
+    let mut reader = BufReader::new(certfile);
+
+    debug!("Loaded certificate {:?}", filename);
+
+    Ok(rustls_pemfile::certs(&mut reader)?
```

Comment:
The documentation should state that only certificates in PEM format are supported.